adds a logo to the README

egpbos opened the pull request:
I generated a simple logo for the template, hope you like it.

vedran-kasalica commented:
I did a bit of manual retouching, @egpbos  what do you think about it? ![python-template copy](https://github.com/user-attachments/assets/9e1379d1-1237-449d-9451-9e81147cb0c0)
I still want to fix the issue with the sneak head and the printer cover.

egpbos commented:
Ha nice, that tongue was bugging me as well :D Let's keep the snake head as it is, that one looks more natural/smooth than the broken tongue. Please commit your logo version to this branch.

vedran-kasalica commented:
> Ha nice, that tongue was bugging me as well :D Let's keep the snake head as it is, that one looks more natural/smooth than the broken tongue. Please commit your logo version to this branch.

I pushed the logo, but I also made a newer version with the head "fixed" (not committed). It looks a bit less AI-generated, not sure if that's the look we are going for. 
![logo-new](https://github.com/user-attachments/assets/2f52e222-ba4b-4363-ae7a-c06447c793c3)

egpbos commented:
I love it, thanks, please push it!

fdiblen commented:
[Our contributing guidelines](https://github.com/NLeSC/python-template/blob/main/CONTRIBUTING.md#you-want-to-make-some-kind-of-change-to-the-code-base) has the following

> (important) announce your plan to the rest of the community before you start working. This announcement should be in the form of a (new) issue;

I suggest we discuss further changes before creating PRs including this one.

fdiblen commented:
[Our contributing guidelines](https://github.com/NLeSC/python-template/blob/main/CONTRIBUTING.md#you-want-to-make-some-kind-of-change-to-the-code-base) has the following

> (important) announce your plan to the rest of the community before you start working. This announcement should be in the form of a (new) issue;

I suggest we discuss the further changes before creating PRs.

vedran-kasalica commented:
I pushed the final version! I also opened an issue #664, related to this PR.

egpbos commented:
Off-topic, but since you brought it up @fdiblen: I think those guidelines are overly strict :) This was not a super high-effort PR. I think discussion could have just taken place here as well, let's not overcomplicate things when it's not necessary.

On-topic: I put some more motivation for this PR in #664, please check it out if you are unsure of the point of this PR and feel free to discuss there before reviewing the PR itself.

fdiblen commented:
> Off-topic, but since you brought it up @fdiblen: I think those guidelines are overly strict :) This was not a super high-effort PR. I think discussion could have just taken place here as well, let's not overcomplicate things when it's not necessary.
> 
> On-topic: I put some more motivation for this PR in #664, please check it out if you are unsure of the point of this PR and feel free to discuss there before reviewing the PR itself.

My point is more about compliance than being strict or amount of work. Let's discuss these in-person and adjust the guidelines to make the team more confortable and efficient. :slightly_smiling_face:

Also, thanks for creating the issue and starting the discussion. :pray:

bouweandela commented:
Is the snake referring to Python? In that case it might be nice to use a python instead of a cobra.

vedran-kasalica commented:
> Is the snake referring to Python? In that case it might be nice to use a python instead of a cobra.

That's a reasonable point, but similarly we can also ask what the printer with a tie represents. Imo there is a certain charm to an AI-generated logo that vaguely depicts the idea behind the project.

On an unrelated note, I was curious about the Photoshop warping feature, so I just played with it a bit, but it is hard to make the body thick without losing quality or spending too much time on it. I am done editing the figure, l promise :)

![logo-python](https://github.com/user-attachments/assets/ce4bbc59-3529-4e23-a608-ee6ffc1adf7c)

egpbos commented:
We don't have to use a python, other snakes are also ecosystem compliant ;) See e.g. anaconda (which is a boa), mamba (a venomous snake, completely different taxonomic family).

In fact, the cobra can "change shape" a bit (flaring its ... flaps?), which could be seen as a symbol of instantiating a template.

Diving into this rabbit hole, I googled for snakes that can change shape or color.

- There is one snake that's known as the "chameleon snake": the [Kapuas mud snake](https://en.wikipedia.org/wiki/Kapuas_mud_snake).
- There are apparently myths/cultural stories about shape changing snakes:
   - Indian folklore has https://en.wikipedia.org/wiki/Ichchadhari_naag
   - A "great Chinese folktale": https://en.wikipedia.org/wiki/Bai_Suzhen

These could also serve as inspiration for #666.

egpbos commented:
Updated with the final version pasted in @vedran-kasalica's last message. Squashed the commits into two to keep the history clean. We had informal approvals and no objections, so I think we can merge now.

egpbos commented:
Rebased on main.
